src/test/java/teammates/test/cases/util/ActivityLogGeneratorTest.java

```diff
@@ -0,0 +1,275 @@
+package teammates.test.cases.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import com.beust.jcommander.internal.Maps;
+import com.google.appengine.api.log.AppLogLine;
+
+import teammates.common.datatransfer.UserType;
+import teammates.common.datatransfer.attributes.AccountAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.PageNotFoundException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.ActivityLogEntry;
+import teammates.common.util.ActivityLogGenerator;
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.logic.api.EmailGenerator;
+import teammates.test.cases.BaseTestCase;
+import teammates.test.driver.AssertHelper;
+
+/**
+ * SUT: {@link ActivityLogGenerator}
+ */
+public class ActivityLogGeneratorTest extends BaseTestCase {
+    
+    private ActivityLogGenerator logCenter = new ActivityLogGenerator();
+
+    // TODO: can move log verification in action test cases to here or remove them totally
```

Comment:
Try not to sprinkle too much TODOs everywhere. They're easily forgotten and you've seen how much we have in the code base right now.